cmd/storage-rest-server.go

```diff
@@ -525,6 +525,48 @@ func (s *storageRESTServer) DeleteFileBulkHandler(w http.ResponseWriter, r *http
 	w.(http.Flusher).Flush()
 }
 
+// DeletePrefixesErrsResp - collection of delete errors
+// for bulk prefixes deletes
+type DeletePrefixesErrsResp struct {
+	Errs []error
+}
+
+// DeletePrefixesHandler - delete a set of a prefixes.
+func (s *storageRESTServer) DeletePrefixesHandler(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		return
+	}
+	vars := r.URL.Query()
+	volume := vars.Get(storageRESTVolume)
+
+	bio := bufio.NewScanner(r.Body)
+	var prefixes []string
+	for bio.Scan() {
+		prefixes = append(prefixes, bio.Text())
+	}
+
+	if err := bio.Err(); err != nil {
+		s.writeErrorResponse(w, err)
+		return
+	}
+
+	errs, err := s.storage.DeletePrefixes(volume, prefixes)
+	if err != nil {
+		s.writeErrorResponse(w, err)
+		return
+	}
+
+	derrsResp := &DeletePrefixesErrsResp{Errs: make([]error, len(errs))}
+	for idx, err := range errs {
+		if err != nil {
+			derrsResp.Errs[idx] = StorageErr(err.Error())
```

Comment:
Since we are deleting the type here, could we just make the struct

```
type DeletePrefixesErrsResp struct {
	Errs []string
}
```
Since we are using GOB, any error type would need to be registered if the type is `error`. That is not a problem with `string`.

It could also be `[]StorageErr`.

Reply:
StorageErr is already registered by gob.